### PR TITLE
Use defaults for ListingOptions builder

### PR DESCRIPTION
**Pull request by mvanschellebeeck**
Follow up PR for https://github.com/apache/arrow-datafusion/issues/4178

### CODE REVIEW

**Comment by mvanschellebeeck** [comment]
Side note: running ` ./dev/rust_lint.sh` locally showed a bunch of clippy errors - filed #4245

**Comment by mvanschellebeeck** [comment]
Tagging @alamb since this is related to the previous ListingOptions PR

File: datafusion/core/src/datasource/listing/table.rs [hunk 1]
```diff
@@ -254,6 +249,26 @@ impl ListingOptions {
             file_sort_order: None,
         }
     }
+
+    /// Build ListingOptions from an existing set of ListingOptions
```
Comment (alamb) [inline datafusion/core/src/datasource/listing/table.rs:253]: I think typically in rust one uses `options.clone()` for this operation

I will leave a comment at the usage site

File: datafusion/core/src/datasource/listing/table.rs [hunk 2]
```diff
@@ -773,10 +788,9 @@ mod tests {
         ];
 
         for (file_sort_order, expected_result) in cases {
-            let options = ListingOptions {
-                file_sort_order,
-                ..options.clone()
-            };
+            let options = ListingOptions::from_options(options.clone())
+                .with_file_sort_order(file_sort_order);
```
Comment (alamb) [inline datafusion/core/src/datasource/listing/table.rs:792]: Would this work? I think we can avoid adding `from_options`

```suggestion
            let options = options.clone()
                .with_file_sort_order(file_sort_order);
```

File: datafusion/core/tests/sql/parquet.rs [hunk 3]
```diff
@@ -57,10 +57,10 @@ async fn parquet_with_sort_order_specified() {
...
 
     // The sort order is not specified
-    let options_no_sort = ListingOptions {
-        file_sort_order: None,
-        ..parquet_read_options.to_listing_options(target_partitions)
-    };
+    let options_no_sort = ListingOptions::from_options(
+        parquet_read_options.to_listing_options(target_partitions),
+    )
+    .with_file_sort_order(None);
```
Comment (alamb) [inline datafusion/core/tests/sql/parquet.rs:63]:
```suggestion
    let options_no_sort = parquet_read_options
      .to_listing_options(target_partitions)
      .with_file_sort_order(None);
```

File: datafusion/core/tests/sql/parquet.rs [hunk 4]
```diff
@@ -72,10 +72,10 @@ async fn parquet_with_sort_order_specified() {
         })
         .collect::<Vec<_>>();
 
-    let options_sort = ListingOptions {
-        file_sort_order: Some(file_sort_order),
-        ..parquet_read_options.to_listing_options(target_partitions)
-    };
+    let options_sort = ListingOptions::from_options(
```
Comment (alamb) [inline datafusion/core/tests/sql/parquet.rs:75]: same comment as above

**Review by alamb — COMMENTED** [review]
Thank you @mvanschellebeeck ! I have a suggetion about `from_options` but otherwise this looks great!

File: datafusion/core/src/datasource/listing/table.rs (hunk 1 above)
Comment (mvanschellebeeck) [reply datafusion/core/src/datasource/listing/table.rs:253]: yep, that makes a lot more sense!

**Review by alamb — APPROVED** [review]
Looks great -- thank you @mvanschellebeeck

**Pull request merged**